### PR TITLE
Use json_response in UserControllerTest

### PR DESCRIPTION
**Pull request by shhavel**
Micro refactoring in `UserControllerTest`:

- json_response helper instead of `Poison.decode!` and assert response status
- used pattern matching, e.g. `%{user: user}` instead of `context`

### CODE REVIEW

File: test/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -21,107 +20,61 @@ defmodule EvercamMedia.UserControllerTest do
...
   end
 
-  test "POST /v1/users when user created successfully!", context do
-    response =
-      build_conn()
-      |> post("/v1/users", context[:params])
+  test "POST /v1/users when user created successfully!", %{params: params} do
+    response = build_conn |> post("/v1/users", params)
 
     assert response.status == 200
```
Comment (shhavel) [inline test/controllers/user_controller_test.exs]: I think for `POST` request successful response status should be 201

**Comment by ijunaidfarooq** [comment]
Nice Effort but these changes are not relevant. We keep it same everywhere.. as We do in other tests...

**Comment by marcoherbst** [comment]
@ijunaidfarooq @shhavel , we've been through this question before. There is some reason why we went for 200, even though 201 is technically correct. (It was more than just about keeping it the same everywhere.) I think it may be because of the defaults in Phoenix. 

@azharmalik3 has had to deal with this already:
http://stackoverflow.com/questions/38953007/phoenix-return-200-status-code-for-post

@azharmalik3 , do you remember the outcome?

**Comment by azharmalik3** [comment]
@marcoherbst yes, I remember.

Actually, phoenix uses default `200` response code for POST. If we want to return `201` code for POST request then must add following `put_status(:created)` in response.

File: test/controllers/user_controller_test.exs (hunk 1 above)
Comment (ijunaidfarooq) [inline test/controllers/user_controller_test.exs]: @shhavel Yes! You are right. We were also thinking of going down the same road. Thanks for your reminder. We are going to implement that from now on.

**Comment by ijunaidfarooq** [comment]
In addition my last comment.

@shankardevy The changes you made in *tests* were good. But we are keeping the tests in the same way as we wrote other ones. Sorry in case you misunderstood. I was referring to the tests, not to the POST status.

**Comment by shhavel** [comment]
@ijunaidfarooq 
Ok, got it. I think at least user update test should be rewritten or moved to model tests,
because there is no HTTP request:

```elixir
test "PATCH /v1/users/ when email is not valid while update!", context do
  params = %{email: "test @test.com"}

  expected_error = %{email: ["Email format isn't valid!"]}

  changeset = User.changeset(context[:user], params)
  error = Util.parse_changeset(changeset)

  assert error == expected_error
end
```

**Comment by ijunaidfarooq** [comment]
@shhavel Yes! You are right am going to do it with HTTP request or may be I will move it. If you may have other suggestions then I will love to hear them too.

**Pull request closed without merging**